Allocator against new isolation service

rostyboost opened the pull request.

gabrielhartmann commented on titus_isolate/allocate/constants.py:
```diff
@@ -26,6 +26,7 @@
 TITUS_ISOLATE_CELL_HEADER = "X-Titus-Isolate-Cell"
 UNKNOWN_CELL = "unknown_cell"
 CELL = "cell"
+GRPC_CELL = "grpc_cell"
```
Why is this needed?  Shouldn't a client land in a particular cell regardless of implementation?  I'd expect the `get_cell` function to determine which API it should address and return the correct cell.

gabrielhartmann commented on titus_isolate/utils.py:
```diff
@@ -181,6 +185,26 @@ def get_cell_name():
         log.error("Failed to determine isolation cell.")
         return UNKNOWN_CELL
 
+def get_grpc_cell_name():
```
Same comment as for the GRPC_CELL tag.

gabrielhartmann commented on titus_isolate/allocate/remote/allocator.py:
```diff
@@ -0,0 +1,164 @@
+ALLOCATOR_NAME = "GrpcRemoteIsolationAllocator"
+REQ_TYPE_METADATA_KEY = "req_type"
+
+class Allocator(CpuAllocator):
```
Can we please name this something a bit less generic?
```
GrpcAllocator
```
maybe

gabrielhartmann commented on titus_isolate/config/constants.py:
```diff
@@ -44,6 +44,11 @@
 REMOTE_REBALANCE_ALLOCATOR = 'REMOTE_REBALANCE_ALLOCATOR'
 
+# New Remote Allocator
+GRPC_REMOTE_ALLOC_ENDPOINT = 'TITUS_ISOLATE_GRPC_REMOTE_ALLOCATOR_ENDPOINT'
```
You probably want all 3 of these in the "restart" list at the bottom of this file.

gabrielhartmann commented on titus_isolate/utils.py:
```diff
@@ -181,6 +185,26 @@ def get_cell_name():
+        log.warning("Config manager is not yet set.")
+        return UNKNOWN_CELL
+
+    endpoint = config_manager.get_str(GRPC_REMOTE_ALLOC_ENDPOINT, None)
```
If this function continues to exist, you probably want `get_cached_str` along with the inclusion in the "restart" list.

gabrielhartmann commented on titus_isolate/allocate/remote/allocator.py:
```diff
@@ -0,0 +1,164 @@
+
+    def __init__(self, free_thread_provider):
+        config_manager = get_config_manager()
+        self.__endpoint = config_manager.get_str(GRPC_REMOTE_ALLOC_ENDPOINT, None)
```
`get_cached_str`

gabrielhartmann commented on titus_isolate/isolate/utils.py:
```diff
@@ -7,11 +7,12 @@
 from titus_isolate.allocate.noop_allocator import NoopCpuAllocator
 from titus_isolate.allocate.noop_reset_allocator import NoopResetCpuAllocator
 from titus_isolate.allocate.remote_cpu_allocator import RemoteCpuAllocator
+from titus_isolate.allocate.remote.allocator import Allocator as GrpcRemoteIsolServiceAllocator
```
This is a 3rd name for the same allocator I think.

gabrielhartmann reviewed (changes requested):
Please see comments.  I assume the core functionality of calling the GRPC service has been verified.  As long as we can return to using the current remote allocator, and dynamically configure the new GRPC remote allocator we should be good to go.

rostyboost commented:
Thanks @gabrielhartmann . I believe I addressed all comments in https://github.com/Netflix-Skunkworks/titus-isolate/pull/337/commits/354dc913a6ede8595ecbde98b99ce0ed693d70d0
